subworkflows/nf-core/fgbio_create_umi_consensus/main.nf

```diff
@@ -0,0 +1,73 @@
+// Convert them to unmapped BAM file, map them to the reference genome,
+// use the mapped information to group UMIs and generate consensus reads
+//
+
+params.fastqtobam_options      = [:]
+params.bam2fq_options          = [args: "-T RX"]
+params.bwamem_options          = [args: "-p -C -M"]
+params.samblaster_options      = [args: "-M --addMateTags", suffix:'_processed']
+params.groupreadsbyumi_options = [:]
+params.groupumistrategy        = "Adjacency"
+params.umiconsensus_options    = [args: '-M 1 -S Coordinate', suffix: '_umiconsensus']
+
+include { FGBIO_FASTQTOBAM                  as FASTQTOBAM }         from '../../../modules/fgbio/fastqtobam/main'                        addParams( options: params.fastqtobam_options )
+include { SAMTOOLS_BAM2FQ                   as BAM2FASTQ }          from '../../../modules/samtools/bam2fq/main.nf'                      addParams( options: params.bam2fq_options )
+include { BWA_INDEX }                                               from '../../../modules/bwa/index/main.nf'                            addParams( options: [:] )
+include { BWA_MEM }                                                 from '../../../modules/bwa/mem/main'                                 addParams( options: params.bwamem_options )
+include { SAMBLASTER }                                              from '../../../modules/samblaster/main'                              addParams( options: params.samblaster_options )
+include { FGBIO_GROUPREADSBYUMI             as GROUPREADSBYUMI }    from '../../../modules/fgbio/groupreadsbyumi/main'                   addParams( options: params.groupreadsbyumi_options )
+include { FGBIO_CALLMOLECULARCONSENSUSREADS as CALLUMICONSENSUS }   from '../../../modules/fgbio/callmolecularconsensusreads/main.nf'    addParams( options: params.umiconsensus_options )
+
+workflow CREATE_UMI_CONSENSUS {
+    take:
+    reads                     // channel: [ val(meta), [ reads ] ]
+    fasta                     // channel: /path/to/reference/fasta
+    read_structure            // channel: val(read_structure)
+
+    main:
+    ch_versions = Channel.empty()
+
+    // using information in val(read_structure) FASTQ reads are converted into
+    // a tagged unmapped BAM file (uBAM)
+    FASTQTOBAM ( reads, read_structure )
+    ch_versions = ch_versions.mix(FASTQTOBAM.out.version)
+
+    // reference is indexed
+    BWA_INDEX ( fasta )
+    ch_versions = ch_versions.mix(BWA_INDEX.out.versions)
+
+    // in order to map uBAM using BWA MEM, we need to convert uBAM to FASTQ
+    // but keep the appropriate UMI tags in the FASTQ comment field and produce
+    // an interleaved FASQT file (hence, split = false)
+    split = false
+    BAM2FASTQ ( FASTQTOBAM.out.umibam, split )
+    ch_versions = ch_versions.mix(BAM2FASTQ.out.versions)
+
+    // appropriately tagged interleaved FASTQ reads are mapped to the reference
+    BWA_MEM ( BAM2FASTQ.out.reads, BWA_INDEX.out.index )
```

Comment:
Maybe it would make sense to directly "connect" here to the mapping workflow, otherwise using BWAMem2 won't work (+ things like having the different sorting types depending if MarkDuplicates or MarkDuplicates Spark is used). One quick idea from would be to have like a pre-mapping_umi subworkflow and post mapping_umi subworkflow or sth like that. Would that be possible?

Reply:
Ah I somehow thought this was a PR to sarek 🤦‍♀️ . But still, my thoughts on not being able to switch out mappers stand

Reply:
I'm thinking to be able to do that, one should include the whole workflow instead of the module? I'm happy to change and test that solution if that's how we can do it :)

Reply:
this one right? https://github.com/nf-core/sarek/blob/dev/subworkflows/nf-core/mapping.nf
can it be ported to modules so the code won't change?

Reply:
right :) I have implemented and tested both aligner using existing modules, in the master branch so there's no sub-workflow dependencies.
run:
```
NF_CORE_MODULES_TEST=1 TMPDIR=/home/lescai/tests/nf-development/outputs/ PROFILE=singularity pytest tests/subworkflows/nf-core/fgbio_create_umi_consensus/ --symlink --wt 2 --kwd
```
and 
```
tests/subworkflows/nf-core/fgbio_create_umi_consensus/test.yml ..............               [100%] 
=== 14 passed in 43.75s ==
```